# Quertermous

## Data
http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE113348

```
   run_accession experiment_accession library_strategy library_source sample_accession          instrument  total_size     run_alias experiment_alias cell.line  size
1     SRR7064074           SRX3994934         ATAC-seq        GENOMIC       SRS3219096         NextSeq 500  2378560051 GSM3111024_r1       GSM3111024      2989  2.22
2     SRR7064073           SRX3994933         ATAC-seq        GENOMIC       SRS3219095         NextSeq 500  7813845377 GSM3111023_r1       GSM3111023      2510  7.28
3     SRR7064072           SRX3994932         ATAC-seq        GENOMIC       SRS3219094         NextSeq 500  3478906012 GSM3111022_r1       GSM3111022      2356  3.24
4     SRR7064070           SRX3994931         ATAC-seq        GENOMIC       SRS3219092         NextSeq 500 10269194012 GSM3111021_r2       GSM3111021      2305  9.56
5     SRR7064069           SRX3994931         ATAC-seq        GENOMIC       SRS3219092         NextSeq 500 10269194012 GSM3111021_r1       GSM3111021      2305  9.56
6     SRR7064071           SRX3994931         ATAC-seq        GENOMIC       SRS3219092         NextSeq 500 10269194012 GSM3111021_r3       GSM3111021      2305  9.56
7     SRR7064068           SRX3994930         ATAC-seq        GENOMIC       SRS3219093         NextSeq 500  4965056660 GSM3111020_r1       GSM3111020    200212  4.62
8     SRR7064067           SRX3994929         ATAC-seq        GENOMIC       SRS3219091         NextSeq 500  2814713627 GSM3111019_r1       GSM3111019      1522  2.62
9     SRR7064065           SRX3994928         ATAC-seq        GENOMIC       SRS3219090         NextSeq 500 10431060788 GSM3111018_r1       GSM3111018      1508  9.71
10    SRR7064066           SRX3994928         ATAC-seq        GENOMIC       SRS3219090         NextSeq 500 10431060788 GSM3111018_r2       GSM3111018      1508  9.71
11    SRR7064064           SRX3994927         ATAC-seq        GENOMIC       SRS3219089         NextSeq 500  4578545558 GSM3111017_r1       GSM3111017      1346  4.26
12    SRR7064063           SRX3994926          RNA-Seq TRANSCRIPTOMIC       SRS3219088 Illumina HiSeq 2500  3479940581 GSM3109114_r1       GSM3109114   1020301  3.24
13    SRR7058339           SRX3989298          RNA-Seq TRANSCRIPTOMIC       SRS3213791 Illumina HiSeq 2500  3837690979 GSM3109165_r1       GSM3109165   9090701  3.57
14    SRR7058338           SRX3989297          RNA-Seq TRANSCRIPTOMIC       SRS3213790 Illumina HiSeq 2500  4378457213 GSM3109164_r1       GSM3109164   9071501  4.08
15    SRR7058337           SRX3989296          RNA-Seq TRANSCRIPTOMIC       SRS3213789         NextSeq 500  2500883546 GSM3109163_r1       GSM3109163   9070202  2.33
16    SRR7058336           SRX3989295          RNA-Seq TRANSCRIPTOMIC       SRS3213788 Illumina HiSeq 2500  8540795515 GSM3109162_r1       GSM3109162   9052004  7.95
17    SRR7058335           SRX3989294          RNA-Seq TRANSCRIPTOMIC       SRS3213787 Illumina HiSeq 2500  3513816291 GSM3109161_r1       GSM3109161   8100901  3.27
18    SRR7058334           SRX3989293          RNA-Seq TRANSCRIPTOMIC       SRS3213786 Illumina HiSeq 2500  5129676350 GSM3109160_r1       GSM3109160   8072501  4.78
19    SRR7058333           SRX3989292          RNA-Seq TRANSCRIPTOMIC       SRS3213785 Illumina HiSeq 2500  3836704550 GSM3109159_r1       GSM3109159   7103002  3.57
20    SRR7058332           SRX3989291          RNA-Seq TRANSCRIPTOMIC       SRS3213784 Illumina HiSeq 2500  5566010053 GSM3109158_r1       GSM3109158  59885590  5.18
21    SRR7058331           SRX3989290          RNA-Seq TRANSCRIPTOMIC       SRS3213783 Illumina HiSeq 2500  6944127703 GSM3109157_r1       GSM3109157  59386145  6.47
22    SRR7058330           SRX3989289          RNA-Seq TRANSCRIPTOMIC       SRS3213782 Illumina HiSeq 2500  4879634815 GSM3109156_r1       GSM3109156    317155  4.54
23    SRR7058329           SRX3989288          RNA-Seq TRANSCRIPTOMIC       SRS3213781 Illumina HiSeq 2500  6593339873 GSM3109155_r1       GSM3109155   3101801  6.14
24    SRR7058328           SRX3989287          RNA-Seq TRANSCRIPTOMIC       SRS3213780 Illumina HiSeq 2500  4535179118 GSM3109154_r1       GSM3109154   3100203  4.22
25    SRR7058327           SRX3989286          RNA-Seq TRANSCRIPTOMIC       SRS3213779 Illumina HiSeq 2500  4293860590 GSM3109153_r1       GSM3109153      3003  4.00
26    SRR7058326           SRX3989285          RNA-Seq TRANSCRIPTOMIC       SRS3213778 Illumina HiSeq 2500  4394846638 GSM3109152_r1       GSM3109152      2999  4.09
27    SRR7058325           SRX3989284          RNA-Seq TRANSCRIPTOMIC       SRS3213777 Illumina HiSeq 2500  4940447601 GSM3109151_r1       GSM3109151      2989  4.60
28    SRR7058324           SRX3989283          RNA-Seq TRANSCRIPTOMIC       SRS3213776 Illumina HiSeq 2500  5100452172 GSM3109150_r1       GSM3109150      2510  4.75
29    SRR7058323           SRX3989282          RNA-Seq TRANSCRIPTOMIC       SRS3213775 Illumina HiSeq 2500  3234056493 GSM3109149_r1       GSM3109149      2463  3.01
30    SRR7058322           SRX3989281          RNA-Seq TRANSCRIPTOMIC       SRS3213774 Illumina HiSeq 2500  4732615585 GSM3109148_r1       GSM3109148      2435  4.41
31    SRR7058321           SRX3989280          RNA-Seq TRANSCRIPTOMIC       SRS3213773 Illumina HiSeq 2500  4555225173 GSM3109147_r1       GSM3109147      2356  4.24
32    SRR7058320           SRX3989279          RNA-Seq TRANSCRIPTOMIC       SRS3213772 Illumina HiSeq 2500  4379281527 GSM3109146_r1       GSM3109146      2315  4.08
33    SRR7058319           SRX3989278          RNA-Seq TRANSCRIPTOMIC       SRS3213771 Illumina HiSeq 2500 12298620613 GSM3109145_r1       GSM3109145      2305 11.45
34    SRR7058318           SRX3989277          RNA-Seq TRANSCRIPTOMIC       SRS3213770 Illumina HiSeq 2500  4762763622 GSM3109144_r1       GSM3109144      2282  4.44
35    SRR7058317           SRX3989276          RNA-Seq TRANSCRIPTOMIC       SRS3213769 Illumina HiSeq 2500  3292642287 GSM3109143_r1       GSM3109143      2228  3.07
36    SRR7058316           SRX3989275          RNA-Seq TRANSCRIPTOMIC       SRS3213768 Illumina HiSeq 2500  3774863453 GSM3109142_r1       GSM3109142      2139  3.52
37    SRR7058315           SRX3989274          RNA-Seq TRANSCRIPTOMIC       SRS3213767 Illumina HiSeq 2500  4008366284 GSM3109141_r1       GSM3109141      2105  3.73
38    SRR7058314           SRX3989273          RNA-Seq TRANSCRIPTOMIC       SRS3213766 Illumina HiSeq 2500  2658595523 GSM3109140_r1       GSM3109140     20805  2.48
39    SRR7058313           SRX3989272          RNA-Seq TRANSCRIPTOMIC       SRS3213765 Illumina HiSeq 2500  2880142375 GSM3109139_r1       GSM3109139   2040401  2.68
40    SRR7058312           SRX3989271          RNA-Seq TRANSCRIPTOMIC       SRS3213764 Illumina HiSeq 2500  3914940951 GSM3109138_r1       GSM3109138   2030801  3.65
41    SRR7058311           SRX3989270          RNA-Seq TRANSCRIPTOMIC       SRS3213763 Illumina HiSeq 2500  6898604395 GSM3109137_r1       GSM3109137    200212  6.42
42    SRR7058310           SRX3989269          RNA-Seq TRANSCRIPTOMIC       SRS3213762 Illumina HiSeq 2500  7096851067 GSM3109136_r1       GSM3109136      1795  6.61
43    SRR7058309           SRX3989268          RNA-Seq TRANSCRIPTOMIC       SRS3213761 Illumina HiSeq 2500  2669514015 GSM3109135_r1       GSM3109135    177089  2.49
44    SRR7058308           SRX3989267          RNA-Seq TRANSCRIPTOMIC       SRS3213759 Illumina HiSeq 2500  4683519471 GSM3109134_r1       GSM3109134      1596  4.36
45    SRR7058307           SRX3989266          RNA-Seq TRANSCRIPTOMIC       SRS3213758 Illumina HiSeq 2500  6464352347 GSM3109133_r1       GSM3109133      1587  6.02
46    SRR7058306           SRX3989265          RNA-Seq TRANSCRIPTOMIC       SRS3213757 Illumina HiSeq 2500  5930605317 GSM3109132_r1       GSM3109132      1576  5.52
47    SRR7058305           SRX3989264          RNA-Seq TRANSCRIPTOMIC       SRS3213756 Illumina HiSeq 2500  4746273700 GSM3109131_r1       GSM3109131      1559  4.42
48    SRR7058304           SRX3989263          RNA-Seq TRANSCRIPTOMIC       SRS3213755 Illumina HiSeq 2500  6350870731 GSM3109130_r1       GSM3109130      1522  5.91
49    SRR7058303           SRX3989262          RNA-Seq TRANSCRIPTOMIC       SRS3213754 Illumina HiSeq 2500  4198153572 GSM3109129_r1       GSM3109129      1508  3.91
50    SRR7058302           SRX3989261          RNA-Seq TRANSCRIPTOMIC       SRS3213753 Illumina HiSeq 2500  6085658716 GSM3109128_r1       GSM3109128      1483  5.67
51    SRR7058301           SRX3989260          RNA-Seq TRANSCRIPTOMIC       SRS3213752 Illumina HiSeq 2500  4590493198 GSM3109127_r1       GSM3109127      1448  4.28
52    SRR7058300           SRX3989259          RNA-Seq TRANSCRIPTOMIC       SRS3213751 Illumina HiSeq 2500  6334933870 GSM3109126_r1       GSM3109126      1401  5.90
53    SRR7058299           SRX3989258          RNA-Seq TRANSCRIPTOMIC       SRS3213750 Illumina HiSeq 2500  4450061502 GSM3109125_r1       GSM3109125      1386  4.14
54    SRR7058298           SRX3989257          RNA-Seq TRANSCRIPTOMIC       SRS3213749 Illumina HiSeq 2500  3989272562 GSM3109124_r1       GSM3109124      1369  3.72
55    SRR7058297           SRX3989256          RNA-Seq TRANSCRIPTOMIC       SRS3213748 Illumina HiSeq 2500  3718945375 GSM3109123_r1       GSM3109123      1347  3.46
56    SRR7058296           SRX3989255          RNA-Seq TRANSCRIPTOMIC       SRS3213747 Illumina HiSeq 2500  3968494619 GSM3109122_r1       GSM3109122      1346  3.70
57    SRR7058295           SRX3989254          RNA-Seq TRANSCRIPTOMIC       SRS3213746 Illumina HiSeq 2500  4241854447 GSM3109121_r1       GSM3109121      1278  3.95
58    SRR7058294           SRX3989253          RNA-Seq TRANSCRIPTOMIC       SRS3213745 Illumina HiSeq 2500  4355340380 GSM3109120_r1       GSM3109120    112201  4.06
59    SRR7058293           SRX3989252          RNA-Seq TRANSCRIPTOMIC       SRS3213744 Illumina HiSeq 2500  4161237329 GSM3109119_r1       GSM3109119     10705  3.88
60    SRR7058292           SRX3989251          RNA-Seq TRANSCRIPTOMIC       SRS3213743 Illumina HiSeq 2500  5111167578 GSM3109118_r1       GSM3109118   1060602  4.76
61    SRR7058291           SRX3989250          RNA-Seq TRANSCRIPTOMIC       SRS3213742 Illumina HiSeq 2500  3321246298 GSM3109117_r1       GSM3109117   1051601  3.09
62    SRR7058290           SRX3989249          RNA-Seq TRANSCRIPTOMIC       SRS3213741 Illumina HiSeq 2500  3456815541 GSM3109116_r1       GSM3109116   1042702  3.22
63    SRR7058289           SRX3989248          RNA-Seq TRANSCRIPTOMIC       SRS3213760 Illumina HiSeq 2500  4624545272 GSM3109115_r1       GSM3109115    102901  4.31
```
## WGS
- Whole-genome sequencing data were processed with the GATK best practices pipeline
- Trimmed Illumina adapters using Cutadapt v1.93
- Trimmed reads were aligned to the hg19 reference genome with Burrows-Wheeler Aligner (BWA) v0.7.124 using its bwa-mem module5 with parameters "-M -t 8 -R '@RG\tID:{sample}\tSM:{sample}\tPL:illumina\tLB:lib1\tPU:unit1'".
- Duplicate reads in alignment result were marked with Picard v1.92. We performed indel
realignment and base recalibration with GATK v3.46
- The GATK HaplotypeCaller was used to generate gVCF files, which were fed into GenotypeGVCFs for joint genotype calling. We recalibrated variants using the GATK VariantRecalibrator module. Since subsequent eQTL calling (see Section 12) with RASQUAL7 required phased variants, we phased our call set with Beagle v4.18
- We first used the Beagle conform-gt module to correct any reference genotypes if
they are different from hg19. We then phased and imputed against 1000 Genomes project
phase 3 version 5a9
- Variants with imputation allelic r2 less than 0.8 and Hardy-Weinberg Equilibrium p-value less than 1x10-6 were filtered out. 

## RNA-Seq
### Prep
Demultiplexing was performed with bcl2fastq script from Illumina.
- Base quality control of demultiplexed sequences was done using FastQC v0.11.4 quality control tool
- Fastq files that correspond to the unique sample were merged using a custom script.
- Mapping of the reads was performed with STAR v2.4.0i11. In accordance to the GATK Best Practices for RNA-Seq we
used the STAR 2-pass alignment pipeline

### Alignment
- First, reads contained in the raw fastq files were mapped to GRCh37/hg19 human genome using STAR and during the first alignment pass splice junctions were discovered with high stringency
- Second pass mapping with STAR was then performed using a new index that was created with splice junction information contained in the file SJ.out.tab from the first pass STAR mapping
- Splice junctions from the first pass were used as annotation in a second pass to permit lower stringency alignment, and therefore higher sensitivity
- Prior to gene expression quantification, we used WASP12 to filter out reads that are prone to mapping bias.
- Read counts and RPKM were calculated with RNA-SeQC v1.1.813 using default parameters with additional flags `-n 1000 -noDoC -strictMode`”` using GENCODE v19 annotation
- Allele-specific read counts were generated with the `createASVCF` module in RASQUAL

### Splicing
We quantified intron excision levels using LeafCutter. In brief, we converted bam files to splice junction files using the `bam2junc.sh` script, and defined intron clusters using leafcutter_cluster.py with default parameters. This requires at least 30 reads supporting each cluster and at least 0.1% of reads supporting each intron within the cluster, and allows intron to have a maximum size of 100kb.

## ATAC-Seq
We used the ENCODE ATAC-seq pipeline to perform alignment and peak calling (https://github.com/kundajelab/atac_dnase_pipelines)
- Steps included in the pipeline
  - FASTQ files were trimmed with Cutadapt v1.93 and aligned with Bowtie2 v2.2.618 with default parameters. Duplicate reads were marked with Picard v1.126.
  - The alignment were converted to ENCODE tagAlign format. Records were shifted +4 and -5 for positive-strand and minus-strand reads.
  - MACS2 v2.0.819 was used to call peaks with default parameters. Each alignment was split into two pseudoreplicate (subsample of reads) and peaks were called independently.
  - Irreproducible Discovery Rate (IDR) 20 analyses were performed based on pseudo-replicates with a cutoff of 0.1 to output an IDR call set, which was used for downstream analysis.
- We used WASP12 to filter out reads that are prone to mapping bias.
